app/assets/javascripts/tramway/multiselect_controller.js

```diff
@@ -0,0 +1,136 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class Multiselect extends Controller {
+  static targets = ["dropdown", "showSelectedArea", "hiddenInput"];
+
+  static values = {
+    items: Array,
+    dropdownContainer: String,
+    itemContainer: String,
+    selectedItemTemplate: String,
+    dropdownState: String,
+    selectedItems: Array,
+    placeholder: String,
+    selectAsInput: String,
+    value: Array
+  }
+
+  connect() {
+    this.dropdownState = 'closed';
+    this.unselectedItems = JSON.parse(this.element.dataset.items);
+
+    const initialValues = this.element.dataset.value === undefined ? [] : this.element.dataset.value.split(',')
+    this.selectedItems = this.unselectedItems.filter(item => initialValues.includes(item.value.toString()));
```

Comment:
can't you do `toString` one time earlier, at the initialization? e.g. like

`this.unselectedItems = JSON.parse(this.element.dataset.items).map(x => x.toString())`

Seems you have many transformation in the code because of that

Reply:
Nice catch! Thanks :slightly_smiling_face: Fixed this and removed all `toString` for unselectedItems.